searches/binarysearch.go

```diff
@@ -5,7 +5,7 @@ func BinarySearch(array []int, target int, lowIndex int, highIndex int) int {
 	if highIndex < lowIndex || len(array) == 0 {
 		return -1
```

Comment:
Consider adding a returning a proper error by changing the return type to include the answer and an error, for example:

```
func BinarySearch(array []int, target int, lowIndex int, highIndex int) (int, error) {
   ...
}

```

and return something like this

```suggestion
return -1, errors.New("The value not found in the array")
```

Reply:
I think -1 is enough to imply not found. Is it not?

Reply:
I agree with @tjgurwara99 on this one.  Delivering an error rather than implying an error is the correct idem in Go.